Allow sshd-auth read network sysctls

zpytela opened the pull request:
The commit addresses the following AVC denial:
type=AVC msg=audit(1764365929.570:676): avc:  denied  { getattr } for  pid=82559 comm="sshd-session" path="/proc/sys/net/ipv6/conf/all/disable_ipv6" dev="proc" ino=15563 scontext=system_u:system_r:sshd_session_t:s0-s0:c0.c1023 tcontext=system_u:object_r:sysctl_net_t:s0 tclass=file permissive=1

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2417769

zpytela committed:
Allow sshd-auth read network sysctls

The commit addresses the following AVC denial:
type=AVC msg=audit(1764365929.570:676): avc:  denied  { getattr } for  pid=82559 comm="sshd-session" path="/proc/sys/net/ipv6/conf/all/disable_ipv6" dev="proc" ino=15563 scontext=system_u:system_r:sshd_session_t:s0-s0:c0.c1023 tcontext=system_u:object_r:sysctl_net_t:s0 tclass=file permissive=1

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2417769

zpytela committed:
Allow sshd-auth capabilities

After dropping privsep patches from openssh, permissions previously
allowed for sshd_net_t now need to be allowed for sshd_auth_t.

The commit addresses the following AVC denials:
type=PROCTITLE msg=audit(12/04/2025 05:56:49.168:81) : proctitle=sshd-auth: [session-auth]
type=PATH msg=audit(12/04/2025 05:56:49.168:81) : item=0 name=/usr/share/empty.sshd inode=157168 dev=fd:02 mode=dir,711 ouid=root ogid=root rdev=00:00 obj=system_u:object_r:usr_t:s0 nametype=NORMAL cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0
type=CWD msg=audit(12/04/2025 05:56:49.168:81) : cwd=/
type=SYSCALL msg=audit(12/04/2025 05:56:49.168:81) : arch=x86_64 syscall=chroot success=yes exit=0 a0=0x55af07e3bfa8 a1=0x55af1af8bd50 a2=0x20 a3=0x55af07e40060 items=1 ppid=1005 pid=1006 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=sshd-auth exe=/usr/libexec/openssh/sshd-auth subj=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 key=(null)
type=AVC msg=audit(12/04/2025 05:56:49.168:81) : avc:  denied  { sys_chroot } for  pid=1006 comm=sshd-auth capability=sys_chroot  scontext=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 tcontext=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 tclass=capability permissive=1

type=PROCTITLE msg=audit(12/04/2025 05:56:49.168:82) : proctitle=sshd-auth: [session-auth]
type=SYSCALL msg=audit(12/04/2025 05:56:49.168:82) : arch=x86_64 syscall=setgroups success=yes exit=0 a0=0x1 a1=0x7ffd7c19d184 a2=0x7ffd7c19d050 a3=0x55af07e3bfea items=0 ppid=1005 pid=1006 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=sshd-auth exe=/usr/libexec/openssh/sshd-auth subj=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 key=(null)
type=AVC msg=audit(12/04/2025 05:56:49.168:82) : avc:  denied  { setgid } for  pid=1006 comm=sshd-auth capability=setgid  scontext=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 tcontext=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 tclass=capability permissive=1

type=PROCTITLE msg=audit(12/04/2025 05:56:49.168:83) : proctitle=sshd-auth: [session-auth]
type=SYSCALL msg=audit(12/04/2025 05:56:49.168:83) : arch=x86_64 syscall=setresuid success=yes exit=0 a0=sshd a1=sshd a2=sshd a3=0x55af07e3c01f items=0 ppid=1005 pid=1006 auid=unset uid=sshd gid=sshd euid=sshd suid=sshd fsuid=sshd egid=sshd sgid=sshd fsgid=sshd tty=(none) ses=unset comm=sshd-auth exe=/usr/libexec/openssh/sshd-auth subj=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 key=(null)
type=AVC msg=audit(12/04/2025 05:56:49.168:83) : avc:  denied  { setuid } for  pid=1006 comm=sshd-auth capability=setuid  scontext=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 tcontext=system_u:system_r:sshd_auth_t:s0-s0:c0.c1023 tclass=capability permissive=1

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2418715

martinpitt commented:
Thanks! This still [fails `passwd -S`](https://artifacts.dev.testing-farm.io/0477f07c-1a40-474a-913a-71fdbd2709ff/work-mainpreqzbt1/plans/cockpit/main/execute/data/guest/default-0/test/browser/main-1/output.txt). Note that we marked that test as "known broken" on rawhide, so the next run will make this green whether or not it actually gets fixed. So that will have to be checked manually (by log inspection or directly checking the command interactively)

zpytela commented:
Taking a look.

zpytela committed:
Allow passwd read and write a sshd-session unnamed pipes

This denial can be triggered e.g. by
ssh hostname passwd -S root

The commit addresses the following AVC denial:
type=PROCTITLE msg=audit(12/05/2025 08:26:47.445:143) : proctitle=passwd -S root
type=PATH msg=audit(12/05/2025 08:26:47.445:143) : item=1 name=/lib64/ld-linux-x86-64.so.2 inode=138318 dev=fd:02 mode=file,755 ouid=root ogid=root rdev=00:00 obj=system_u:object_r:ld_so_t:s0 nametype=NORMAL cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0
type=PATH msg=audit(12/05/2025 08:26:47.445:143) : item=0 name=/usr/bin/passwd inode=139481 dev=fd:02 mode=file,suid,755 ouid=root ogid=root rdev=00:00 obj=system_u:object_r:passwd_exec_t:s0 nametype=NORMAL cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0
type=EXECVE msg=audit(12/05/2025 08:26:47.445:143) : argc=3 a0=passwd a1=-S a2=root
type=SYSCALL msg=audit(12/05/2025 08:26:47.445:143) : arch=x86_64 syscall=execve success=yes exit=0 a0=0x55f20882da30 a1=0x55f20882d100 a2=0x55f208833030 a3=0x8 items=2 ppid=1127 pid=1128 auid=root uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=3 comm=passwd exe=/usr/bin/passwd subj=unconfined_u:unconfined_r:passwd_t:s0-s0:c0.c1023 key=(null)
type=AVC msg=audit(12/05/2025 08:26:47.445:143) : avc:  denied  { write } for  pid=1128 comm=passwd path=pipe:[11848] dev="pipefs" ino=11848 scontext=unconfined_u:unconfined_r:passwd_t:s0-s0:c0.c1023 tcontext=system_u:system_r:sshd_session_t:s0-s0:c0.c1023 tclass=fifo_file permissive=0
type=AVC msg=audit(12/05/2025 08:26:47.445:143) : avc:  denied  { write } for  pid=1128 comm=passwd path=pipe:[11847] dev="pipefs" ino=11847 scontext=unconfined_u:unconfined_r:passwd_t:s0-s0:c0.c1023 tcontext=system_u:system_r:sshd_session_t:s0-s0:c0.c1023 tclass=fifo_file permissive=0
type=AVC msg=audit(12/05/2025 08:26:47.445:143) : avc:  denied  { read } for  pid=1128 comm=passwd path=pipe:[11846] dev="pipefs" ino=11846 scontext=unconfined_u:unconfined_r:passwd_t:s0-s0:c0.c1023 tcontext=system_u:system_r:sshd_session_t:s0-s0:c0.c1023 tclass=fifo_file permissive=0

zpytela commented:
@martinpitt With the latest build, ssh passwd -S works and does not report a denial.

martinpitt commented:
Thanks @zpytela !
